yoyodyne/models/modules/transformer.py

```diff
@@ -73,6 +73,26 @@ def forward(
         return out
 
 
+class AttentionOutput:
+    """Tracks an attention output during the forward pass.
+    
+    This object can be passed into a hook to the Transformer forward pass
+    in order to modify it's behavior such that certaina ttention weights are 
+    returned, and then stored in self.outputs."""
+    def __init__(self):
+        self.outputs = []
+
+    def __call__(self, module, module_in, module_out):
+        """Stores the second return argument of `module`.
+        
+        This is intended to be called on a multiehaded attention, which returns
+        both the contextualized representation, and the attention weights."""
+        self.outputs.append(module_out[1])
+
+    def clear(self):
+        self.outputs = []
```

Comment:
Why not just `self.outputs.clear()`? Probably slightly cheaper, certainly clearer to me.